### PR TITLE
[SYCL][ESIMD] Add conversion ctor and operator

### PR DESCRIPTION
**Pull request by cmc-rep**
Signed-off-by: Gang Y Chen <gang.y.chen@intel.com>

### CODE REVIEW

**Comment by cmc-rep** [comment]
@kbobrovs, a test failed. However, after looking into the detail, the failure seems not relevant to esimd. What should I do in this case?

**Comment by cmc-rep** [comment]
@vladimirlaz, jenkins test failed. However, looking at the error details, I don't think it is related to this PR. How can we resolve this?

**Comment by kbobrovs** [comment]
Looks like infrastructure failure. I restarted the build. I think every PR submitter should be able to do this - click 'details' for the web page where rebuild can be started

File: sycl/test/esimd/simd.cpp [hunk 1]
```diff
@@ -212,14 +220,12 @@ bool test_replicate2() __attribute__((sycl_device)) {
...
+  return v0_rep[0] == v0[1] && v0_rep[1] == v0[2] && v0_rep[2] == v0[5];
 }
 
 bool test_replicate3() __attribute__((sycl_device)) {
   simd<int, 8> v0(0, 1);
   auto v0_rep = v0.replicate<2, 4, 2, 2>(1);
 
-  return v0_rep[0] == v0[1] && v0_rep[1] == v0[3] &&
-         v0_rep[2] == v0[5];
+  return v0_rep[0] == v0[1] && v0_rep[1] == v0[3] && v0_rep[2] == v0[5];
```
Comment (kbobrovs) [inline sycl/test/esimd/simd.cpp:230]: please add a testcase for `%=`

are you going to create executable tests for new esimd functionality you add?

Comment (cmc-rep) [reply sycl/test/esimd/simd.cpp:231]: test added. Yes, I will modify some existing examples to use the new esimd functionality.

Comment (cmc-rep) [reply sycl/test/esimd/simd.cpp:231]: @kychendev @kbobrovs please review and merge timely

**Review by kychendev — APPROVED** [review]

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**